---
title: "R Notebook - R Assignment BCB 546X"
output: html_notebook
---

#**Important - If you are recreating this code, you will need to change the paths for each of the files listed if you have cloned the data to your computer.** 
#Part 1

#First I need to import the two files I will be using. To do this I will use read_delim().
#Using view() also allows me to look at the file as well.
#Depending on where the original files are saved, you will need to change the path of the files.

```{r}
library(readr)
fang <- read_delim("C:/Users/Joe_S/Desktop/BCB Class/R_Assignment/fang_et_al_genotypes.txt", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
View(fang)
```

#There's a reason I did col_names = FALSE on the genotype file and not on the snp file next. I wanted to keep a header for the snp file, however, I did not want a header for the genotype file. The reason for this is because I will need to transpose the two separate genotype files I will need later, but need the potential header from the genotype file to be transposed and a portion of the dataset also later on. The snp file is fine as is and the header will be used later on.

```{r}
library(readr)
snp <- read_delim("C:/Users/Joe_S/Desktop/BCB Class/R_Assignment/snp_position.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
View(snp)
```

#Now I want look at the structure of both datasets using structure().
#I do this to identify similarities between the two files.

```{r}
str(fang)
head(snp)
```

#From this I can see I have mostly character variables and some interger variables within the snp file.

#Now I want to separate out my genotype file by group type.
#I would like a file for maize only and for teosinte only.

```{r}
maize <- fang[fang$X3 == "Group" | fang$X3 == "ZMMIL" | fang$X3 == "ZMMLR" | fang$X3 == "ZMMMR",]
teosinte <- fang[fang$X3 == "Group" | fang$X3 == "ZMPBA" | fang$X3 == "ZMPIL" | fang$X3 == "ZMPJA",]
```

#I had to keep Group in both searches, as if I didn't it would have eliminated the SNP_ID's that I will need once I transpose the data to be able to merge.

#As before to be able to merge both datasets together I need to transpose my genotype files.

```{r}
t_maize <- t(maize)
head(t_maize)
t_teosinte <- t(teosinte)
head(t_teosinte)
```

#Now that I see the data is arranged approriately, I want to add back the column names as the first row of data and I do not want to keep the first 3 lines of the data frame from both genotype files.

```{r}
colnames(t_maize) <- t_maize[1,]
t_maize <- t_maize[-1:-3,]
head(t_maize)
NROW(t_maize)

colnames(t_teosinte) <- t_teosinte[1,]
t_teosinte <- t_teosinte[-1:-3,]
head(t_teosinte)
NROW(t_teosinte)
```

#Now I have a snp file and two genotype files (maize and teosinte) that match in length at 983 rows and a common column between the two, but I need to rename the genotype column "Sample_ID" to match the snp file as "SNP_ID".

```{r}
colnames(t_maize)[1] <- "SNP_ID"
colnames(t_teosinte)[1] <- "SNP_ID"
```

#Now I can merge the two files together to produce one file for maize and one file for teosinte.

```{r}
maize_merged <- merge(snp, t_maize, by = "SNP_ID")
teosinte_merged <- merge(snp, t_teosinte, by = "SNP_ID")

head(maize_merged)
head(teosinte_merged)
```

#To make sure my columns are correct according to the assignment, I will switch the 2nd column to the 4th column location, to have SNP_ID, Chromosome and Position in order.

```{r}
maize_merged <- maize_merged[c(1,3,4,2,5:1588)]
head(maize_merged)
teosinte_merged <- teosinte_merged[c(1,3,4,2,5:990)]
head(teosinte_merged)
```

#Now I want to split and sort by position, but I have some with position as multiple, unknown or NA. To be able to perform the next step without excluding those I will categorize multiple positions as 0 and unknowns to also be NA. Disregard the chromosome information, I am attempting to run my chromosome and sorting through a for loop function and need to match chromosome to numeric. Still can't get it to work.

```{r}
maize_merged$Position[maize_merged$Position == "multiple"] <- 0
maize_merged$Position[maize_merged$Position == "unknown"] <- NA
maize_merged$Chromosome[maize_merged$Chromosome == "multiple"] <- 0
maize_merged$Chromosome[maize_merged$Chromosome == "unknown"] <- NA

teosinte_merged$Position[teosinte_merged$Position == "multiple"] <- 0
teosinte_merged$Position[teosinte_merged$Position == "unknown"] <- NA
teosinte_merged$Chromosome[teosinte_merged$Chromosome == "multiple"] <- 0
teosinte_merged$Chromosome[teosinte_merged$Chromosome == "unknown"] <- NA
```

#I will split the maize file into files by chromosome and I want to sort by position. However, looking back at the structure of the position variable, it is a character and not a value and will not sort correctly. I must change the structure of the position variable and then sort.

```{r}
maize_merged$Position = as.numeric(as.character(maize_merged$Position))
maize_merged$Chromosome = as.numeric(as.character(maize_merged$Chromosome))
teosinte_merged$Position = as.numeric(as.character(teosinte_merged$Position))
teosinte_merged$Chromosome = as.numeric(as.character(teosinte_merged$Chromosome))
```

#Now that I have changed the position variable to numeric, I can split my files into the 10 chromosome files and apply the other criteria such as ascending or descending order or substituting values for unknown SNP's. I then output the files I create using the write.table function and specify the path. If someone else is using this code they will need to change the path to their own folder of choice.

#Maize Chromosome 1
```{r}
maize_1 = maize_merged[maize_merged$Chromosome == 1,]
maize_1 = maize_1[order(maize_1$Position),]
maize_1_ascend = data.frame(lapply(maize_1, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_1_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_1_ascend.txt", sep = "\t")
maize_1 = maize_1[order(rev(maize_1$Position)),]
maize_1_descend = data.frame(lapply(maize_1, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_1_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_1_descend.txt", sep = "\t")
```
#Maize Chromosome 2
```{r}
maize_2 <- maize_merged[maize_merged$Chromosome == 2,]
maize_2 <- maize_2[order(maize_2$Position),]
maize_2_ascend = data.frame(lapply(maize_2, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_2_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_2_ascend.txt", sep = "\t")
maize_2 = maize_2[order(rev(maize_2$Position)),]
maize_2_descend = data.frame(lapply(maize_2, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_2_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_2_descend.txt", sep = "\t")
```
#Maize Chromosome 3
```{r}
maize_3 <- maize_merged[maize_merged$Chromosome == 3,]
maize_3 <- maize_3[order(maize_3$Position),]
maize_3_ascend = data.frame(lapply(maize_3, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_3_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_3_ascend.txt", sep = "\t")
maize_3 = maize_3[order(rev(maize_3$Position)),]
maize_3_descend = data.frame(lapply(maize_3, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_3_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_3_descend.txt", sep = "\t")
```
#Maize Chromosome 4
```{r}
maize_4 <- maize_merged[maize_merged$Chromosome == 4,]
maize_4 <- maize_4[order(maize_4$Position),]
maize_4_ascend = data.frame(lapply(maize_4, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_4_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_4_ascend.txt", sep = "\t")
maize_4 = maize_4[order(rev(maize_4$Position)),]
maize_4_descend = data.frame(lapply(maize_4, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_4_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_4_descend.txt", sep = "\t")
```
#Maize Chromosome 5
```{r}
maize_5 <- maize_merged[maize_merged$Chromosome == 5,]
maize_5 <- maize_5[order(maize_5$Position),]
maize_5_ascend = data.frame(lapply(maize_5, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_5_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_5_ascend.txt", sep = "\t")
maize_5 = maize_5[order(rev(maize_5$Position)),]
maize_5_descend = data.frame(lapply(maize_5, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_5_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_5_descend.txt", sep = "\t")
```
#Maize Chromosome 6
```{r}
maize_6 <- maize_merged[maize_merged$Chromosome == 6,]
maize_6 <- maize_6[order(maize_6$Position),]
maize_6_ascend = data.frame(lapply(maize_6, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_6_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_6_ascend.txt", sep = "\t")
maize_6 = maize_6[order(rev(maize_6$Position)),]
maize_6_descend = data.frame(lapply(maize_6, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_6_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_6_descend.txt", sep = "\t")
```
#Maize Chromosome 7
```{r}
maize_7 <- maize_merged[maize_merged$Chromosome == 7,]
maize_7 <- maize_7[order(maize_7$Position),]
maize_7_ascend = data.frame(lapply(maize_7, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_7_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_7_ascend.txt", sep = "\t")
maize_7 = maize_7[order(rev(maize_7$Position)),]
maize_7_descend = data.frame(lapply(maize_7, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_7_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_7_descend.txt", sep = "\t")
```
#Maize Chromosome 8
```{r}
maize_8 <- maize_merged[maize_merged$Chromosome == 8,]
maize_8 <- maize_8[order(maize_8$Position),]
maize_8_ascend = data.frame(lapply(maize_8, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_8_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_8_ascend.txt", sep = "\t")
maize_8 = maize_8[order(rev(maize_8$Position)),]
maize_8_descend = data.frame(lapply(maize_8, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_8_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_8_descend.txt", sep = "\t")
```
#Maize Chromosome 9
```{r}
maize_9 <- maize_merged[maize_merged$Chromosome == 9,]
maize_9 <- maize_9[order(maize_9$Position),]
maize_9_ascend = data.frame(lapply(maize_9, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_9_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_9_ascend.txt", sep = "\t")
maize_9 = maize_9[order(rev(maize_9$Position)),]
maize_9_descend = data.frame(lapply(maize_9, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_9_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_9_descend.txt", sep = "\t")
```
#Maize Chromosome 10
```{r}
maize_10 <- maize_merged[maize_merged$Chromosome == 10,]
maize_10 <- maize_10[order(maize_10$Position),]
maize_10_ascend = data.frame(lapply(maize_10, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(maize_10_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_10_ascend.txt", sep = "\t")
maize_10 = maize_10[order(rev(maize_10$Position)),]
maize_10_descend = data.frame(lapply(maize_10, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(maize_10_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/maize_10_descend.txt", sep = "\t")
```


#Teosinte Chromosome 1
```{r}
teosinte_1 = teosinte_merged[teosinte_merged$Chromosome == 1,]
teosinte_1 = teosinte_1[order(teosinte_1$Position),]
teosinte_1_ascend = data.frame(lapply(teosinte_1, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_1_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_1_ascend.txt", sep = "\t")
teosinte_1 = teosinte_1[order(rev(teosinte_1$Position)),]
teosinte_1_descend = data.frame(lapply(teosinte_1, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_1_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_1_descend.txt", sep = "\t")
```
#Teosinte Chromosome 2
```{r}
teosinte_2 <- teosinte_merged[teosinte_merged$Chromosome == 2,]
teosinte_2 <- teosinte_2[order(teosinte_2$Position),]
teosinte_2_ascend = data.frame(lapply(teosinte_2, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_2_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_2_ascend.txt", sep = "\t")
teosinte_2 = teosinte_2[order(rev(teosinte_2$Position)),]
teosinte_2_descend = data.frame(lapply(teosinte_2, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_2_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_2_descend.txt", sep = "\t")
```
#Teosinte Chromosome 3
```{r}
teosinte_3 <- teosinte_merged[teosinte_merged$Chromosome == 3,]
teosinte_3 <- teosinte_3[order(teosinte_3$Position),]
teosinte_3_ascend = data.frame(lapply(teosinte_3, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_3_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_3_ascend.txt", sep = "\t")
teosinte_3 = teosinte_3[order(rev(teosinte_3$Position)),]
teosinte_3_descend = data.frame(lapply(teosinte_3, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_3_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_3_descend.txt", sep = "\t")
```
#Teosinte Chromosome 4
```{r}
teosinte_4 <- teosinte_merged[teosinte_merged$Chromosome == 4,]
teosinte_4 <- teosinte_4[order(teosinte_4$Position),]
teosinte_4_ascend = data.frame(lapply(teosinte_4, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_4_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_4_ascend.txt", sep = "\t")
teosinte_4 = teosinte_4[order(rev(teosinte_4$Position)),]
teosinte_4_descend = data.frame(lapply(teosinte_4, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_4_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_4_descend.txt", sep = "\t")
```
#Teosinte Chromosome 5
```{r}
teosinte_5 <- teosinte_merged[teosinte_merged$Chromosome == 5,]
teosinte_5 <- teosinte_5[order(teosinte_5$Position),]
teosinte_5_ascend = data.frame(lapply(teosinte_5, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_5_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_5_ascend.txt", sep = "\t")
teosinte_5 = teosinte_5[order(rev(teosinte_5$Position)),]
teosinte_5_descend = data.frame(lapply(teosinte_5, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_5_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_5_descend.txt", sep = "\t")
```
#Teosinte Chromosome 6
```{r}
teosinte_6 <- teosinte_merged[teosinte_merged$Chromosome == 6,]
teosinte_6 <- teosinte_6[order(teosinte_6$Position),]
teosinte_6_ascend = data.frame(lapply(teosinte_6, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_6_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_6_ascend.txt", sep = "\t")
teosinte_6 = teosinte_6[order(rev(teosinte_6$Position)),]
teosinte_6_descend = data.frame(lapply(teosinte_6, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_6_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_6_descend.txt", sep = "\t")
```
#Teosinte Chromosome 7
```{r}
teosinte_7 <- teosinte_merged[teosinte_merged$Chromosome == 7,]
teosinte_7 <- teosinte_7[order(teosinte_7$Position),]
teosinte_7_ascend = data.frame(lapply(teosinte_7, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_7_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_7_ascend.txt", sep = "\t")
teosinte_7 = teosinte_7[order(rev(teosinte_7$Position)),]
teosinte_7_descend = data.frame(lapply(teosinte_7, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_7_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_7_descend.txt", sep = "\t")
```
#Teosinte Chromosome 8
```{r}
teosinte_8 <- teosinte_merged[teosinte_merged$Chromosome == 8,]
teosinte_8 <- teosinte_8[order(teosinte_8$Position),]
teosinte_8_ascend = data.frame(lapply(teosinte_8, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_8_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_8_ascend.txt", sep = "\t")
teosinte_8 = teosinte_8[order(rev(teosinte_8$Position)),]
teosinte_8_descend = data.frame(lapply(teosinte_8, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_8_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_8_descend.txt", sep = "\t")
```
#Teosinte Chromosome 9
```{r}
teosinte_9 <- teosinte_merged[teosinte_merged$Chromosome == 9,]
teosinte_9 <- teosinte_9[order(teosinte_9$Position),]
teosinte_9_ascend = data.frame(lapply(teosinte_9, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_9_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_9_ascend.txt", sep = "\t")
teosinte_9 = teosinte_9[order(rev(teosinte_9$Position)),]
teosinte_9_descend = data.frame(lapply(teosinte_9, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_9_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_9_descend.txt", sep = "\t")
```
#Teosinte Chromosome 10
```{r}
teosinte_10 <- teosinte_merged[teosinte_merged$Chromosome == 10,]
teosinte_10 <- teosinte_10[order(teosinte_10$Position),]
teosinte_10_ascend = data.frame(lapply(teosinte_10, function(x){gsub("\\?\\/\\?", "\\?", x)}))
write.table(teosinte_10_ascend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_10_ascend.txt", sep = "\t")
teosinte_10 = teosinte_10[order(rev(teosinte_10$Position)),]
teosinte_10_descend = data.frame(lapply(teosinte_10, function(x){gsub("\\?\\/\\?", "\\-", x)}))
write.table(teosinte_10_descend, "C:/Users/Stock/Desktop/BCB-546X/R_Assignment/teosinte_10_descend.txt", sep = "\t")
```

#Part 2

#Since the data loaded in from the previous exercise still exists I want to create a total merged file since we will be looking at all groups now and then install the reshape2 package.

```{r}
install.packages("reshape2")
library(reshape2)
fang_melt <- melt(fang)
snp_melt <- melt(snp)
```

#I will graph the number of SNPs per Chromosome in the next part.

```{r}
library(ggplot2)
num_snps <- as.data.frame(table(snp$Chromosome))
num_snps
num_snps <- num_snps[c(1,3:10,2,11:12),]
num_snps$Var1 <- as.character(num_snps$Var1)
num_snps$Var1 <- factor(num_snps$Var1, levels=unique(num_snps$Var1))
ggplot(num_snps, aes(Var1,Freq)) + geom_point() + xlab("Chromosome") + ylab("Number of SNPs") + ggtitle("Number of SNPs per Chromosome") + theme(plot.title = element_text(hjust = 0.5))
```

#As shown in the graph, SNP # is decreasing from Chromosome 1 - 10. SNP's mapped to multiple chromosomes and unknown location are added at the end to identify how many are included in those categories as well. 

#The next step I want to take is to look at group number and identify how many samples each group is contributing to these numbers.

```{r}
num_groups <- as.data.frame(table(fang$X3))
num_groups
num_groups <- num_groups[-1,]
ggplot(num_groups, aes(Var1,Freq)) + geom_point() + xlab("Group") + ylab("Number of Observations") + ggtitle("Number of Observations per Group") + theme(plot.title = element_text(hjust = 0.5))
```

#I need to add the header back to the fang dataset to look at the individual SNPs question from the homework.

```{r}
colnames(fang) <- fang[1,]
fang <- fang[-1,]
```

```{r}
fang$abph1.20_num <- ""
fang$abph1.20_num[fang$abph1.20 == "C/C"] <- 2
fang$abph1.20_num[fang$abph1.20 == "G/G"] <- 2
fang$abph1.20_num[fang$abph1.20 == "C/G"] <- 1
fang$abph1.20_num[fang$abph1.20 == "G/C"] <- 1
fang$abph1.20_num[fang$abph1.20 == "?/?"] <- NA
```

#This has converted the homozygous to 2 and the heterozygous to 1 and ?/? to NA.
#Now I want to plot the frequency of the SNPs in each of the groups.

```{r}
ggplot(fang, aes(x=Group, fill = abph1.20_num)) + geom_bar(aes(y = (..count..)/sum(..count..)),position = position_dodge(width = 0.5)) + scale_y_continuous(breaks = c(0,0.05,0.12,0.2,0.25,0.3,0.35)) + ylab("Frequency") + ggtitle("Homozygous vs Heterozygous Frequency in the abph1.20 SNP by Group") + theme(plot.title = element_text(hjust = 0.5))
```

#As can be seen in the graph there are a few groups that have the large numbers. However, I can also see that in all cases the homozygous SNPs are in greater proportion to the heterozygous SNPs. In one group (TRIP) this SNP is not present. I used posiiton_dodge within the graph to stagger the histogram as well.


